Kudu.Core/Deployment/DeploymentHelper.cs

```diff
@@ -49,5 +51,46 @@ public static bool IsDefaultWebRootContent(string webroot)
 
             return false;
         }
+
+        public static void PurgeZipsIfNecessary(string sitePackagesPath, ITracer tracer)
+        {
+            int DEFAULT_PURGE_VAL = 5;
+            int MIN_PURGE_VAL = 1;
+
+            var totalAllowedZips = DEFAULT_PURGE_VAL;
+            if (System.Environment.GetEnvironmentVariable("PURGE_VAL") != null)
+            {
+                try
+                {
+                    totalAllowedZips = Int32.Parse(System.Environment.GetEnvironmentVariable("PURGE_VAL"));
+                    totalAllowedZips = totalAllowedZips < MIN_PURGE_VAL ? MIN_PURGE_VAL : totalAllowedZips;
+                }
+                catch (Exception ex)
+                {
+                    tracer.TraceError(ex, "Exception thrown while retrieving Environment variable value. Invalid value set in the Environment variable.");
+                    totalAllowedZips = DEFAULT_PURGE_VAL;
+                }
+            }
+            IEnumerable<string> zipFiles = FileSystemHelpers.GetFiles(sitePackagesPath, "*.zip");
+            if (zipFiles.Count() > totalAllowedZips)
+            {
+                // Create a list of files and their last mod date in a keypair and sort in descending order.
+                var fileModDateList = new List<KeyValuePair<string, DateTime>>(zipFiles.Select(fileName => new KeyValuePair<string, DateTime>(fileName, FileSystemHelpers.GetLastWriteTimeUtc(fileName))));
```

Comment:
This code can probably be simplified by doing an OrderBy on the collection, and avoid the need for a temporary list to do the sorting.

Reply:
You are right! I will update it right away. Thanks!

Reply:
You can also do `.Skip(totalAllowedZips)` on the enumerable, and then delete everything that's left (assuming descending sort).

Reply:
Cool, I didn't know `.Skip(int)` existed. Thanks!